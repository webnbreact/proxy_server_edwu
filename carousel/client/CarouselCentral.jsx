import React, { Component } from 'react';
import { BrowserRouter as Brouter, Route } from 'react-router-dom';
import CarouselLanding from './components/CarouselLanding';
import Modal from './components/Modal';

// eslint-disable-next-line react/prefer-stateless-function
class CarouselCentral extends Component {
  // Brouters only allow one child element
  constructor() {
    super();
    this.state = {
      isModal: false,
      imageUrls: [],
      highlightedPicIdx: 1,
    };
    this.handleModalClick = this.handleModalClick.bind(this);
    this.handleGetPictures = this.handleGetPictures.bind(this);
  }

  handleModalClick(e = null, highlightedPicIdx = 0) {
    e.preventDefault();
    const { isModal } = this.state;
    console.log(highlightedPicIdx);
    this.setState({
      isModal: !isModal,
      highlightedPicIdx,
    });
  }

  handleGetPictures(imageUrls) {
    this.setState({
      imageUrls: imageUrls,
    });
  }

  render() {
    const { isModal, imageUrls, highlightedPicIdx } = this.state;

    return (
      <div style={{ width: '90%' }}>
        <CarouselLanding
          highlightedPicIdx={highlightedPicIdx}
          picAmt={imageUrls.length}
          inModal={isModal}
          handleGetPictures={(imageUrls) => { this.handleGetPictures(imageUrls); }}
          handleModalClick={this.handleModalClick}
        />

        {isModal && (
          <Modal
            highlightedPicIdx={highlightedPicIdx}
            picAmt={imageUrls.length}
            inModal={isModal}
            imageUrls={imageUrls}
            handleModalClick={this.handleModalClick}
          />
        )
        }
      </div>
    );
  }
}

export default CarouselCentral;
