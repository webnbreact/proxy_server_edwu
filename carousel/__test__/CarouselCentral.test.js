import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import CarouselCentral from '../client/CarouselCentral';

Enzyme.configure({ adapter: new Adapter() });

// describe('hey', () => {

// });

describe('Carousel Central', () => {
  it('should exist', () => {
    const tree = shallow(<CarouselCentral />);
    expect(toJson(tree)).toMatchSnapshot();
  });
});

// test('This tests how many nodes are on the screen', () => {
//   const tree = shallow(<CarouselLanding />);
// });
