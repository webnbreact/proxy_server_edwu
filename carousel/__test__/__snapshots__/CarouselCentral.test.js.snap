// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel Central should exist 1`] = `
<div>
  <CarouselLanding />
</div>
`;
